oms-azure-storage-analytics-solution/scripts/AzureSACreateSchedules-MS-Mgmt-SA.ps1

```diff
@@ -24,9 +29,69 @@ do
 		Start-Sleep -s 60        
 	}
 } while (-not $syncOk -and $retry -ge 0)
+"Selecting Azure subscription..."
+$SelectedAzureSub = Select-AzureRmSubscription -SubscriptionId $ArmConn.SubscriptionId -TenantId $ArmConn.tenantid 
+#Creating headers for REST ARM Interface
+$subscriptionid=$ArmConn.SubscriptionId
+"Azure rm profile path  $((get-module -Name AzureRM.Profile).path) "
+$path=(get-module -Name AzureRM.Profile).path
+$path=Split-Path $path
+$dlllist=Get-ChildItem -Path $path  -Filter Microsoft.IdentityModel.Clients.ActiveDirectory.dll  -Recurse
+$adal =  $dlllist[0].VersionInfo.FileName
+try
+{
+	Add-type -Path $adal
+	[reflection.assembly]::LoadWithPartialName( "Microsoft.IdentityModel.Clients.ActiveDirectory" )
+}
+catch
+{
+	$ErrorMessage = $_.Exception.Message
+	$StackTrace = $_.Exception.StackTrace
+	Write-Warning "Error during sync: $ErrorMessage, stack: $StackTrace. "
+}
+#Create authentication token using the Certificate for ARM connection
+$certs= Get-ChildItem -Path Cert:\Currentuser\my -Recurse | Where{$_.Thumbprint -eq $ArmConn.CertificateThumbprint}
+#$certs
+[System.Security.Cryptography.X509Certificates.X509Certificate2]$mycert=$certs[0]
+#Write-output "$mycert will be used to acquire token"
+$CliCert=new-object   Microsoft.IdentityModel.Clients.ActiveDirectory.ClientAssertionCertificate($ArmConn.ApplicationId,$mycert)
+$AuthContext = new-object Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext("https://login.windows.net/$($ArmConn.tenantid)")
+$result = $AuthContext.AcquireToken("https://management.core.windows.net/",$CliCert)
```

Comment:
this won't work in anything but the AzureCloud environment, can we fix this?

Reply:
removed [reflection.assembly]::LoadWithPartialName